### PR TITLE
v2: add DNSResource

### PR DESCRIPTION
**Pull request by oxtoacart**
Updates tailscale/corp#21867

### CODE REVIEW

File: v2/dns.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	}))
+	if err != nil {
+		return err
+	}
+
+	return dr.do(req, nil)
+}
+
+// SearchPaths retrieves the list of search paths that is currently set for the given tailnet.
+func (dr *DNSResource) SearchPaths(ctx context.Context) ([]string, error) {
```
Comment (mpminardi) [inline v2/dns.go:37]: Have noticed this elsewhere, but to be explicit: we're going with the pattern of the GET operation being just the name of the thing we are getting when it doesn't interact with a top level object (e.g., top level like a `Devices().List()`) vs prepending the operation to have something like `ListSearchPaths`?

Comment (oxtoacart) [reply v2/dns.go:37]: Put differently, there is no url like `dns/searchpaths/{id}`.

Comment (oxtoacart) [reply v2/dns.go:37]: And one other way to put it - `searchPaths` behaves like a property of the `DNS` resource, as opposed to a collection of its own.

Comment (mpminardi) [reply v2/dns.go:37]: Ah gotcha, treating it as a property on the `DNS` resource makes sense to me. I don't have super strong opinions myself but was trying to make sense of our pattern here in my head so thank you for walking through the thinking!

**Review by mpminardi — APPROVED** [review]

**Pull request merged**